src/datasources/privat24/privat24.js

```diff
@@ -1,3 +1,209 @@
+    availableBalance: cardBalance.av_balance._text,
+    dateBalance: cardBalance.bal_date._text,
+    creditLimit: cardBalance.fin_limit._text
+  };
+  resolve(result);
+};
+
+const getData = (dataForRequest, cb) => {
+  const path = cb.name === 'handleTransactionsData' ? '/p24api/rest_fiz' :
+    cb.name === 'handleBalanceData' ? '/p24api/balance' :
+      null;
+  const options = {
+    hostname: 'api.privatbank.ua',
+    path,
+    method: 'POST',
+    headers:
+        { 'Content-Type': 'application/xml; charset=UTF-8' }
+  };
+  const req = http.request(options);
+  const xml = cb.name === 'handleTransactionsData' ?
+    fillTransactionsXml(dataForRequest) :
+    cb.name === 'handleBalanceData' ? fillBalanceXml(dataForRequest) :
+      null;
+
+  const signature = createSignature(xml, dataForRequest.merchantPassword);
+  const xmlWithSignature = fillXmlWithSignature(xml, signature);
+  req.write(xmlWithSignature);
+  req.end();
+
+  return new Promise((resolve, reject) => {
+    req.on('response', handleBankResponse.bind(null, cb, resolve, reject));
+    req.on('error', err => reject(err));
+  });
+};
+
+const getTransactionsData = async dataForTransactions => {
+  const startDate = stringToDate(dataForTransactions.startDate);
+  const endDate = stringToDate(dataForTransactions.endDate);
+
+  const periods = divideIntoPeriods(startDate, endDate);
+  const promises = [];
+
+  periods.forEach((period, index) => {
+    const copyDataForTransactions = Object.assign({}, dataForTransactions);
+    copyDataForTransactions.startDate = dateToString(period[0]);
+    copyDataForTransactions.endDate = dateToString(period[1]);
+    promises[index] = getData(copyDataForTransactions, handleTransactionsData);
+  });
+  const transactions = await Promise.all(promises);
+
+  let result = [];
+  for (const array of transactions) result = [...result, ...array.reverse()];
+
+  return result;
+};
+
+const getBalanceData = async dataForBalance => {
+  const result = [];
+  for (const data of dataForBalance) {
+    result.push(getData(data, handleBalanceData));
+  }
+  return Promise.allSettled(Object.values(result));
+};
+
+const dataForTransactionsRequest = { // sample data for transactions request
+  merchantPassword: '55x3Ft9C96yx7s1cAMO2KVn1apuDA0X6',
+  merchantId: 123456,
+  wait: 10,
+  test: 0,
+  paymentId: '',
+  startDate: '11.05.2020',
+  endDate: '23.10.2020',
+  cardNumber: '1234567890123456'
+};
+
+const dataForBalanceRequest = { // sample data for balance request
+  merchantPassword: '55x3Ft9C96yx7s1cAMO2KVn1apuDA0X6',
+  merchantId: 123456,
+  wait: 10,
+  test: 0,
+  paymentId: '',
+  cardNumber: '1234567890123456',
+  country: 'UA'
+};
+
+// example of using
+(async () => {
+  const tranData = await getTransactionsData(dataForTransactionsRequest);
+  const balanceData = await getBalanceData([dataForBalanceRequest]);
+  // dataForBalanceRequest is array of data objects
+  console.log(tranData);
+  console.log(balanceData);
+})();
```

Comment:
Move it to another file, it shouldn't been executed every time we use this file

Reply:
Ok, i will just put this in comment